swarm/pss/pss_test.go

```diff
@@ -482,29 +486,32 @@ func testAsymSend(t *testing.T) {
 	}
 }
 
-// params in run name:
-// nodes/msgs/addrbytes/adaptertype
-// if adaptertype is exec uses execadapter, simadapter otherwise
-func TestNetwork(t *testing.T) {
-	//t.Run("8/1024/2/exec", testNetwork)
-	t.Run("256/128/4/sock", testNetwork)
+type Job struct {
+	Msg      []byte
+	SendNode discover.NodeID
+	RecvNode discover.NodeID
 }
 
-func testNetwork(t *testing.T) {
+func worker(id int, jobs <-chan Job, rpcs map[discover.NodeID]*rpc.Client, pubkeys map[discover.NodeID][]byte, hextopic string) {
+	for j := range jobs {
+		rpcs[j.SendNode].Call(nil, "pss_sendAsym", common.ToHex(pubkeys[j.RecvNode]), hextopic, j.Msg)
+	}
+}
 
+func TestNetwork(t *testing.T) {
 	type msgnotifyC struct {
 		id     discover.NodeID
 		msgIdx int
 	}
 	topic := whisper.BytesToTopic([]byte("foo:42"))
 	hextopic := common.ToHex(topic[:])
 
-	paramstring := strings.Split(t.Name(), "/")
-	nodecount, _ := strconv.ParseInt(paramstring[1], 10, 0)
-	msgcount, _ := strconv.ParseInt(paramstring[2], 10, 0)
-	addrsize, _ := strconv.ParseInt(paramstring[3], 10, 0)
-	messagedelayvarianceusec := (int(msgcount)/1000 + 1) * 1000 * 1000
-	log.Info("network test", "nodecount", nodecount, "msgcount", msgcount, "addrhintsize", addrsize, "sendtimevariance", messagedelayvarianceusec/(1000*1000))
+	nodecount := *fnodecount
+	msgcount := *fmsgs
+	addrsize := *faddrsize
+	adapter := *fadapter
+
```

Comment:
Ultimately, the test should be run with different combinations of parameters, if the parameters aren't explicitly set on the command line. To my understanding of go, using `testing.T.Run()` is the better way to do this, because it ensures the environment is properly clean for each run?

Reply:
I found it easier to have the command line flags, when running the test in different combinations, rather than edit the configuration all the time. If we want to run multiple combinations as part of the test suite, then this is not going to work.

I don't have strong feelings about how this should work, I can easily revert the previous version.

Reply:
I believe we should do both, as described: overridable with args, but by default a suite of params to test sane performance.